cdap-explore/src/main/java/co/cask/cdap/explore/guice/ExploreRuntimeModule.java

```diff
@@ -256,21 +258,39 @@ private static void setupClasspath() throws IOException {
     // It could even be wrong to keep them because in the target container, the boot class path may be different
     // (for example, if Hadoop uses a different Java version than CDAP).
 
-    Set<String> bootstrapClassPaths = ExploreServiceUtils.getBoostrapClasses();
+    final Set<String> bootstrapClassPaths = ExploreServiceUtils.getBoostrapClasses();
+
+    ClassAcceptor classAcceptor = new ClassAcceptor() {
+       /* Excluding any class contained in the bootstrapClassPaths and Kryo classes and hive-exec.jar
+        * We need to remove Kryo dependency in the Explore container. Spark introduced version 2.21 version of Kryo,
+        * which would be normally shipped to the Explore container. Yet, Hive requires Kryo 2.22,
+        * and gets it from the Hive jars - hive-exec.jar to be precise.
+        * hive-exec is the job.jar in hive, since hive has guava-11.0.2 classes
+        * we want to exclude the hive-exec.jar from classpath.
+        * */
+      @Override
+      public boolean accept(String className, URL classUrl, URL classPathUrl) {
+        if (bootstrapClassPaths.contains(classPathUrl.getFile()) ||
+          className.startsWith("com.esotericsoftware.kryo") || classPathUrl.getFile().contains("hive-exec")) {
```

Comment:
In distributed mode, actually we can exclude all transitive dependencies that brought in by Spark. We don't have to change the code node, but can you open a JIRA for that?